### PR TITLE
Fix Pow hint

### PR DESCRIPTION
**Pull request by Juan-M-V**

### CODE REVIEW

File: pkg/hints/pow_hints.go [hunk 1]
```diff
@@ -10,13 +10,15 @@ import (
...
 // %{ ids.locs.bit = (ids.prev_locs.exp % PRIME) & 1 %}
 func pow(ids IdsManager, vm *VirtualMachine) error {
-	prev_locs_exp_addr, err := ids.GetAddr("prev_locs", vm)
-	prev_locs_exp, _ := vm.Segments.Memory.GetFelt(prev_locs_exp_addr.AddUint(4))
+	prev_locs_exp_addr, err := ids.GetRelocatable("prev_locs", vm)
+	if err != nil {
+		return err
+	}
 
+	prev_locs_exp, err := vm.Segments.Memory.GetFelt(prev_locs_exp_addr.AddUint(4))
```
Comment (fmoletta) [inline pkg/hints/pow_hints.go:18]:
```suggestion
	prev_locs_exp_addr, err := ids.GetStructFieldFelt("prev_locs", 4, vm)
```

File: pkg/hints/pow_hints_test.go [hunk 2]
```diff
@@ -9,10 +9,11 @@ import (
 	"testing"
 )
 
-func TestPowHintOk(t *testing.T) {
+func TestPowHintOddOk(t *testing.T) {
 	vm := NewVirtualMachine()
 	vm.Segments.AddSegment()
-	vm.Segments.Memory.Insert(NewRelocatable(0, 4), NewMaybeRelocatableFelt(FeltFromUint64(5)))
+	vm.Segments.Memory.Insert(NewRelocatable(0, 4), NewMaybeRelocatableFelt(FeltFromUint64(3)))
+
```
Comment (fmoletta) [inline pkg/hints/pow_hints_test.go:17]: SetupIdsForTest supports structs so you can do this instead:
```
idsManager := SetupIdsForTest(
		map[string][]*MaybeRelocatable{
			"prev_locs": {nil, nil, nil, nil, NewMaybeRelocatableFelt(FeltFromUint64(3))},
```

File: pkg/hints/pow_hints.go (hunk 1 above)
Comment (Juan-M-V) [reply pkg/hints/pow_hints.go:18]: I've been trying to make this work for a while, but I wasn't able to. I think there is a dereference somewhere that might not be handled.
Here is how `prev_locs` is initialized in the cairo function that calls the hint:
https://github.com/starkware-libs/cairo-lang/blob/27a157d761ae49b242026bcbe5fca6e60c1e98bd/src/starkware/cairo/common/pow.cairo#L25

File: pkg/hints/pow_hints_test.go (hunk 2 above)
Comment (Juan-M-V) [reply pkg/hints/pow_hints_test.go:17]: This also fails when trying to use it on the tests

File: pkg/hints/pow_hints.go (hunk 1 above)
Comment (fmoletta) [reply pkg/hints/pow_hints.go:18]: Oh sorry, my bad, I thought it said GetAddr instead of GetRelocatable

**Review by fmoletta — APPROVED** [review]

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**